src/transformers/models/beit3/convert_beit3_unilm_to_pytorch.py

```diff
@@ -0,0 +1,371 @@
+    if "beit3_base_patch16_224_in1k" in checkpoint_url:
+        output = model(pixel_values=torch.tensor(input["pixel_values"]))
+        if validate_logits:
+            assert output.logits.shape == torch.Size([1, 1000])
+            np.testing.assert_allclose(
+                output.logits.detach().numpy()[:, :3], torch.tensor([[-0.260473, -0.420061, -0.492118]]), rtol=1e-05
+            )
+    elif "beit3_base_patch16_480_vqa" in checkpoint_url:
+        output = model(
+            input_ids=torch.tensor(input["input_ids"]),
+            pixel_values=torch.tensor(input["pixel_values"]),
+            text_padding_mask=torch.ones(input["input_ids"].shape),
+        )
+        if validate_logits:
+            assert output.logits.shape == torch.Size([1, 3129])
+            np.testing.assert_allclose(
+                output.logits.detach().numpy()[:, :3], torch.tensor([[-10.862484, -12.388088, -7.6599636]]), rtol=1e-05
+            )
+    elif "beit3_base_patch16_224_nlvr2" in checkpoint_url:
+        pixel_values = torch.cat(
+            (torch.tensor(input["pixel_values"]).unsqueeze(1), torch.tensor(input["pixel_values"]).unsqueeze(1)), dim=1
+        )
+        output = model(
+            input_ids=torch.tensor(input["input_ids"]),
+            pixel_values=pixel_values,
+            text_padding_mask=torch.ones(input["input_ids"].shape),
+        )
+        if validate_logits:
+            assert output.logits.shape == torch.Size([1, 2])
+            np.testing.assert_allclose(
+                output.logits.detach().numpy(), torch.tensor([[6.593818, -6.582055]]), rtol=1e-05
+            )
+    elif "beit3_base_patch16_480_coco_captioning" in checkpoint_url:
+        language_masked_pos = torch.zeros(input["input_ids"].shape)
+        to_fill = list(range(0, input["input_ids"].shape[1], 3))
+        language_masked_pos[:, to_fill] = 1
+        output = model(
+            input_ids=torch.tensor(input["input_ids"]),
+            pixel_values=torch.tensor(input["pixel_values"]),
+            text_padding_mask=torch.ones(input["input_ids"].shape),
+            language_masked_pos=language_masked_pos,
+        )
+        if validate_logits:
+            assert output.logits.shape == torch.Size([3, 64010])
+            np.testing.assert_allclose(
+                output.logits.detach().numpy()[0, :3], torch.tensor([-19.36987, -19.369905, -17.022049]), rtol=1e-05
+            )
+    elif "beit3_base_patch16_384_coco_retrieval" in checkpoint_url:
+        another_input_ids = beit3_processor(text=["This is photo of a dog"], images=image)["input_ids"]
+        output = model(
+            input_ids=torch.tensor([input["input_ids"][0], another_input_ids[0]]),
+            pixel_values=torch.tensor([input["pixel_values"][0], input["pixel_values"][0]]),
+        )
+        if validate_logits:
+            assert round(float(output.loss.detach().numpy()), 4) == 1.8435
+
+    Path(pytorch_dump_folder_path).mkdir(exist_ok=True)
```

Comment:
pretty sure save pretrained takes care of this

Reply:
I saw it in all the conversion scripts, hence kept it .